### ISSUE TITLE
feature request: include git binary in builder image

### ISSUE THREAD

**Issue by evanshortiss** [sclorg/s2i-nodejs-container#279]
I recently ran into an issue where my builds failed. It was due to the fact I was using installing a package I had forked and updated on GitHub. In other words, I was installing the package from GitHub. A `package.json` entry for this looks similar to the following JSON:

```js
"dependencies": {
  "env-var": "github:username/repository#branch"
}
```

You can quickly reproduce this issue like so:

```bash
# create reproducible failing project
mkdir /tmp/s2i-node-git
cd /tmp/s2i-node-git
npm init -f 
npm i evanshortiss/env-var#master -S

# important, must delete local modules or they copied into the build
# meaning it will succeed since the git dep already exists :)
rm -rf node_modules

# build using the local copy of the code and nodes ubi s2i image
s2i build -c . registry.access.redhat.com/ubi8/nodejs-14 s2i-node-git

# also happens using rhel image
s2i build -c . registry.redhat.io/rhel8/nodejs-14 s2i-node-git-fail
```

The following logs are produced:

```
s2i build -c . registry.redhat.io/rhel8/nodejs-14 s2i-node-git-fail
---> Installing application source ...
---> Installing all dependencies
npm WARN s2i-node-git@1.0.0 No description
npm WARN s2i-node-git@1.0.0 No repository field.

npm ERR! code ENOENT
npm ERR! syscall spawn git
npm ERR! path git
npm ERR! errno -2
npm ERR! enoent Error while executing:
npm ERR! enoent undefined ls-remote -h -t ssh://git@github.com/evanshortiss/env-var.git
npm ERR! enoent
npm ERR! enoent
npm ERR! enoent spawn git ENOENT
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent
```

This indicates that Git was not found. The Dockerfile used by this builder *does* [appear to install Git](https://github.com/sclorg/s2i-nodejs-container/blob/master/14/Dockerfile.rhel8#L57-L58), but I might be misunderstanding it somehow.

**Comment by evanshortiss** [comment]
Using the following building and running Dockerfile also indicates Git is not installed:

```
FROM registry.redhat.io/rhel8/nodejs-14

CMD ["git"]
```

**Comment by pkubatrh** [comment]
Hi @evanshortiss
This is already fixed via #274, it just takes a while for the change to propagate to the images built by Red Hat.

**Comment by evanshortiss** [comment]
Ah, thanks! I should have searched closed issues too 😊
